include/zephyr/linker/common-ram.ld

```diff
@@ -141,6 +141,10 @@
 	ITERABLE_SECTION_RAM(zbus_channel_observation_mask, 1)
 #endif /* CONFIG_ZBUS */
 
+#if defined(CONFIG_DEVICE_MUTABLE)
+	ITERABLE_SECTION_RAM(device_mutable, 4)
+#endif
```

Comment:
nit: you need to update linker generator cmake files as well.

Reply:
Fixed!